### PR TITLE
Mf 281 select course content

### PR DESCRIPTION
**Pull request by ChughAkshay**

### CODE REVIEW

File: src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs [hunk 1]
```diff
@@ -289,14 +290,42 @@ private string GenerateLimitReachedBackLink(ReservationsRouteModel routeModel)
         [HttpPost]
         [ValidateAntiForgeryToken]
         [Route("{id}/select-course", Name = RouteNames.EmployerSelectCourse)]
-        public async Task<IActionResult> PostSelectCourse(ReservationsRouteModel routeModel, string selectedCourseId)
+        public async Task<IActionResult> PostSelectCourse(ReservationsRouteModel routeModel, PostSelectCourseViewModel postViewModel)
         {
+            if (!ModelState.IsValid)
+            {
+                var cachedReservation = await _mediator.Send(new GetCachedReservationQuery { Id = routeModel.Id.Value });
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs]: This part is the same as above in the get, can we move this into a BuildEmployerSelectCourseViewModel method to remove the duplication of code?

File: src/SFA.DAS.Reservations.Web/Models/PostSelectCourseViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Reservations.Web.Models
+{
+    public class PostSelectCourseViewModel
+    {
+        public string SelectedCourseId { get; set; }
+        [Required(ErrorMessage = "Select whether you know which apprenticeship training your apprentice will take")]
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Web/Models/PostSelectCourseViewModel.cs]: Not sure if this is needed anymore

Comment (ChughAkshay) [reply src/SFA.DAS.Reservations.Web/Models/PostSelectCourseViewModel.cs]: change pushed

File: src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs (hunk 1 above)
Comment (ChughAkshay) [reply src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs]: change pushed

**Pull request merged**